src/compiler/types.ts

```diff
@@ -1740,12 +1740,14 @@ namespace ts {
         readonly kind: SyntaxKind.NullKeyword;
     }
 
-    export interface TrueLiteral extends PrimaryExpression {
+    export interface TrueLiteral extends LiteralExpression {
         readonly kind: SyntaxKind.TrueKeyword;
+        readonly text: never
     }
 
-    export interface FalseLiteral extends PrimaryExpression {
+    export interface FalseLiteral extends LiteralExpression {
         readonly kind: SyntaxKind.FalseKeyword;
+        readonly text: never

```

Comment:
I'm not a big fan of this. Maybe it would be better to change `isLiteralExpression` to type guard to a supertype of `LiteralExpression`? Presumably callers will want to access `text`, and should know that it's not defined in some cases.

Reply:
I'm also not a fan of this. We use `LiteralExpression` to mean things that have a possibly differentiating literal value represented as a sequence of unicode characters (i.e., string, number, bigint, RegExp literal). This also disregards `null` as being a "literal" value. We perhaps could introduce a super-type shared between `LiteralExpression` and these keyword expressions, but I would not support a change that introduced an unused `text` property.